pkgs/tools/misc/fclones/default.nix

```diff
@@ -0,0 +1,30 @@
+{ lib, stdenv, rustPlatform, fetchFromGitHub, pkg-config, udev }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "fclones";
+  version = "0.14.0";
+
+  src = fetchFromGitHub {
+    owner = "pkolaczk";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1ni5amy903cy822yhw070bcrrixrw2m1vr66q1h32bc98pyv4w05";
+  };
+
+  cargoSha256 = "1gcb46k7bwdfsf6hyvmi6dna1nf6myzy63bhjfp0wy7c8g4m2mg8";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ udev ];
+
+  # tests in dedupe.rs fail due to
+  # "creation time is not available for the filesystem"
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Efficient Duplicate File Finder and Remover";
+    homepage = "https://github.com/pkolaczk/fclones";
+    license = licenses.mit;
+    maintainers = with maintainers; [ cyounkins ];
+  };
+}
+
```

Comment:
```suggestion
```

Reply:
Is there a style guide I'm missing? If there is I'll adhere to it.

I thought it was good practice to have a newline at the end of a file due to POSIX standards - https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Reply:
that looks like an extra new line to me, github usually displays files without a newline differently

Reply:
I find it remarkably confusing the way different editors present newlines: https://stackoverflow.com/questions/16222530/why-do-i-need-vim-in-binary-mode-for-noeol-to-work

Here's what hexdump thinks the end of the file looks like:
```
0000310 7d20 0a3b 0a7d 000a
0000317
```

So is that a NUL and then an LF or a 4-byte encoded LF? I don't know but I don't think it matters.

Reply:
fiddled with `hexdump` with `set binary noeol` for a while, i don't think `hexdump` show trailing newlines as hex numbers

this is without newline
```
0000310 7d20 0a3b 007d
0000315
```

this is with trailing newline
```
0000310 7d20 0a3b 0a7d
0000316
```

this is with an extra newline
```
0000310 7d20 0a3b 0a7d 000a
0000317
```

Reply:
nixpkgs-fmt can be used to format files https://github.com/nix-community/nixpkgs-fmt

Reply:
@Artturin Thanks! I'll start using that.

`nixpkgs-fmt` doesn't care about any number of trailing newlines. I'm still confused on what editors are doing and if hexdump could be hiding some bytes from me, but in the interest of closing this out I'm pushing an update with no blank line at the end. I think we're good to go!